packages/sn-document-viewer-react/src/components/DocumentViewerRegeneratePreviews.tsx

```diff
@@ -40,10 +41,14 @@ export const DocumentViewerRegeneratePreviews: React.FC = () => {
         </Typography>
         <Button
           variant="contained"
-          disabled={isRegenerating}
-          onClick={() => {
+          disabled={isRegenerating || isInProgress}
+          onClick={async () => {
             setIsRegenerating(true)
-            api.regeneratePreviews({ document: documentData, abortController: new AbortController() })
+            const result = await api.regeneratePreviews({
+              document: documentData,
+              abortController: new AbortController(),
+            })
+            updateDocumentData({ ...documentData, pageCount: result.PageCount })
```

Comment:
setIsRegenerating(false) is missing (currently this component is unmounted after the request, but it is a potential bug for the future)